$MirOS$

[Pine-info] Using the attachment part name as description

--- pine/mailview.c.orig	Tue Sep 20 18:25:58 2005
+++ pine/mailview.c	Mon May  8 15:45:09 2006
@@ -1711,8 +1711,16 @@ format_message(msgno, env, body, handles
 	    if(n = strlen(a->size))
 	      strncpy(tmp + size_offset - n, a->size, n);
 
-	    if((n = ps_global->ttyo->screen_cols - (size_offset + 4)) > 0)
+	    if((n = ps_global->ttyo->screen_cols - (size_offset + 4)) > 0) {
 	      strncpy(tmp + size_offset + 2, a->description, n);
+	      n -= (2 + strlen(a->description));
+	      if (n > 2 && a->body && a->body->parameter
+		  && (tmp1 = rfc2231_get_param(a->body->parameter, "name",
+					       NULL, NULL))) {
+		    strcat(tmp, ": ");
+	            strncat(tmp, tmp1, n);
+	      }
+	    }
 
 	    tmp[ps_global->ttyo->screen_cols - (margin[0] + margin[1])] = '\0';
 
